lib/bundler/definition.rb

```diff
@@ -228,12 +232,13 @@ def requested_specs
     end
 
     def current_dependencies
-      dependencies.select(&:should_include?)
+      dependencies.select do |d|
+        d.should_include? && !d.gem_platforms(@platforms).empty?
```

Comment:
`!d.gem_platforms(@platforms).empty?` is the same check done in `expand_dependencies` that removes dependencies not for the target platforms. For example, `nokogiri` gem is removed for `gem "nokogiri", :platforms => [:mingw, :mswin, :x64_mingw, :jruby]` with `forge_ruby_platform` configuration in `spec/runtime/platform_spec.rb:107`.